pkg/controller/clusterserviceversion/clusterserviceversion_controller.go

```diff
@@ -196,5 +278,98 @@ func (r *ReconcileClusterServiceVersion) Reconcile(request reconcile.Request) (r
 	}
 
 	reqLogger.Info("reconcilation complete")
-	return reconcile.Result{}, nil
+	return reconcile.Result{RequeueAfter: time.Minute * 1}, nil
+}
+
+// deleteExternalResources searches for the MeterDefinition created by the CSV, if it's found delete it
+func (r *ReconcileClusterServiceVersion) deleteExternalResources(CSV *olmv1alpha1.ClusterServiceVersion) error {
+	reqLogger := log.WithValues("Request.Name", CSV.GetName(), "Request.Namespace", CSV.GetNamespace())
+	reqLogger.Info("deleting csv")
+	var err error
+
+	meterDefinitionList := &marketplacev1alpha1.MeterDefinitionList{}
+	err = r.client.List(context.TODO(), meterDefinitionList, client.InNamespace(CSV.GetNamespace()))
+	if err != nil {
+		return err
+	}
+
+	// search specifically for the Meter Definition created by the CSV
+	for _, meterDefinition := range meterDefinitionList.Items {
```

Comment:
Meter definitions may be created by different CSVs. This would delete any meter def with this annotation.